packages/backend/src/apps/telegram-bot/common/interceptor/rate-limit.ts

```diff
@@ -0,0 +1,32 @@
+import type { IApp } from '@plumber/types'
+
+import logger from '@/helpers/logger'
+
+const rateLimitHandler: IApp['requestErrorHandler'] = async function (
+  $,
+  error,
+) {
+  if (error.response.status !== 429) {
+    return // use default error handling
+  }
+
+  const retryAfter = +error.response.headers['retry-after']
+  if (isNaN(retryAfter)) {
+    return
+  }
+
+  logger.error('Telegram rate limit error', {
+    executionId: $.execution.id,
+    flowId: $.flow.id,
+    stepId: $.step.id,
+    data: error.response.data,
+    headers: error.response.headers,
+    oldRetryAfter: retryAfter,
+    newRetryAfter: retryAfter + 30,
```

Comment:
nit: should store `newRetryAfter`  in a separate const.  Can we also increase to ~1min / 60 or 70 sec since official rate limit is 20 messages per minute to same group. So we guarantee that we make progress each time.

Reply:
noob qns but can i just check to store as a separate const cus it is used again later?